src/y-webrtc.js

```diff
@@ -466,96 +466,136 @@ const openRoom = (doc, provider, name, key) => {
+    rooms.forEach(room =>
+      publishSignalingMessage(this, room, { type: 'announce', from: room.peerId })
+    )
+  }
+
+  handleMessage (roomName, data) {
+    const room = rooms.get(roomName)
+    if (room == null || typeof roomName !== 'string') {
+      return
+    }
+    const execMessage = data => {
+      const webrtcConns = room.webrtcConns
+      const peerId = room.peerId
+      if (data == null || data.from === peerId || (data.to !== undefined && data.to !== peerId) || room.bcConns.has(data.from)) {
+        // ignore messages that are not addressed to this conn, or from clients that are connected via broadcastchannel
+        return
+      }
+      const emitPeerChange = webrtcConns.has(data.from)
+        ? () => {}
+        : () =>
+          room.provider.emit('peers', [{
+            removed: [],
+            added: [data.from],
+            webrtcPeers: Array.from(room.webrtcConns.keys()),
+            bcPeers: Array.from(room.bcConns)
+          }])
+      switch (data.type) {
+        case 'announce':
+          if (webrtcConns.size < room.provider.maxConns) {
+            map.setIfUndefined(webrtcConns, data.from, () => new WebrtcConn(this, true, data.from, room))
+            emitPeerChange()
           }
-          if (room.key) {
-            if (typeof m.data === 'string') {
-              cryptoutils.decryptJson(buffer.fromBase64(m.data), room.key).then(execMessage)
+          break
+        case 'signal':
+          if (data.signal.type === 'offer') {
+            const existingConn = webrtcConns.get(data.from)
+            if (existingConn) {
+              const remoteToken = data.token
+              const localToken = existingConn.glareToken
+              if (localToken && localToken > remoteToken) {
+                log('offer rejected: ', data.from)
+                return
+              }
+              // if we don't reject the offer, we will be accepting it and answering it
+              existingConn.glareToken = undefined
             }
-          } else {
-            execMessage(m.data)
           }
-        }
+          if (data.signal.type === 'answer') {
+            log('offer answered by: ', data.from)
+            const existingConn = webrtcConns.get(data.from)
+            existingConn.glareToken = undefined
+          }
+          if (data.to === peerId) {
+            map.setIfUndefined(webrtcConns, data.from, () => new WebrtcConn(this, false, data.from, room)).peer.signal(data.signal)
+            emitPeerChange()
+          }
+          break
       }
-    })
-    this.on('disconnect', () => log(`disconnect (${url})`))
+    }
+    if (room.key) {
+      if (typeof data === 'string') {
+        cryptoutils.decryptJson(buffer.fromBase64(data), room.key).then(execMessage)
+      }
+    } else {
+      execMessage(data)
+    }
+  }
+
+  connected () {
+    return this.client.connected
+  }
+
+  subscribe (roomName) {
+    this.client.send({ type: 'subscribe', topics: [roomName] })
```

Comment:
These functions ends up doing the same as the originals, right ? uses the send method ... so why replacing them ...